pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/IntermediateSegment.java

```diff
@@ -103,19 +100,6 @@ public IntermediateSegment(SegmentGeneratorConfig segmentGeneratorConfig) {
       }
     }
 
-    SegmentPartitionConfig segmentPartitionConfig = segmentGeneratorConfig.getSegmentPartitionConfig();
```

Comment:
`IntermediateSegment` is not anywhere other than one test but I have still fixed considering some might use it  in future.